ledger/src/blockstore_processor.rs

```diff
@@ -127,6 +172,22 @@ fn execute_batch(
             timings,
         );
 
+    let post_process_units: u64 = aggregate_total_execution_units(timings);
+    let execution_cost_units: u64 = post_process_units - pre_process_units;
+    let remaining_block_cost_cap = cost_capacity_meter
+        .write()
+        .unwrap()
+        .accumulate(execution_cost_units);
+    debug!(
+        "batch of {} transactions, total execute cu {}, remaining block cost cap {}",
+        batch.sanitized_transactions().len(),
+        execution_cost_units,
+        remaining_block_cost_cap
+    );
+    if remaining_block_cost_cap == 0_u64 {
+        return Err(TransactionError::WouldExceedMaxBlockCostLimit);
+    }
+
```

Comment:
This should be gated behind a feature so that validators don't fork off

Reply:
Yea, it'd be safer to featurize it. Will make the change.